### ISSUE TITLE
exceedingly weird auto-completion for en_GB

### ISSUE THREAD

**Issue by Cathryne** [Zinggi/DictionaryAutoComplete#19]
This can not really be explained, only shown ;-) Please copy-paste the below instructions into a new document and follow the steps. I used [ST3b3083](https://www.sublimetext.com/3) with the en_GB dictionary on both Win7 64bit and Mac OS X 10.11.1. The behavior described below stops only when I  disable/remove DictionaryAutoComplete.

```
- 1a) move cursor in front of hyphen; 1b) press tab => inserts tab
    - 2a) move cursor between tab and hyphen; 2b) press tab => auto-completion replaces hypen with 1st word in dictionary (use-defined 
 - 3) same effect with whitespace(s) instead of tab before hyphen
```

I hope whichever [dictionary](https://packagecontrol.io/packages/Dictionaries) you use to test this has a less extreme word at the top than the one I noticed this bug with (en_GB) ;->

**Comment by Zinggi** [comment]
That's a very interesting observation, I was a bit surprised when `al-Qa'ida` suddenly popped up in my text ;)

What's happening is:
Sublime Text issues an `'insert best completion'` command when you hit tab. Usually you don't have any words with a `-` in your possible suggestion, so nothing gets inserted and Sublime inserts an actual tab character instead. Since this package provides autocomplete suggestions with all words from your dictionary file, the list of possible suggestions is quite long and Sublime inserts the "best", e.g. the one on the top of that list.
If you keep pressing tab, you'll notice that you get other words such as `T-junction` and `T-shirt`.

Unfortunately, I can only explain what's happening, but I can't disable that behavior.
This is a flaw in the way the sublime text autocompletion system works and I have no control over that.

I tried if the same thing also happens if I'm editing a code file and indeed it does.
I installed a clojure package (since lisp uses `-` inside function names), disabled DictionaryAutoComplete and indeed, pressing tab in front of a `-` produces some seemingly random function name with a dash in it.
